test/t8_schemes/t8_gtest_successor.cxx

```diff
@@ -83,82 +103,64 @@ t8_deep_successor (t8_element_t *element, t8_element_t *successor,
-    /*Go to the children at maximum level */
+  int                 num_children = ts->t8_element_num_children (element);
+  int                 num_children_child;
+
+  for (int ichild = 0; ichild < num_children; ichild++) {
+    ts->t8_element_child (element, ichild, child);
+    /* Go to the children at maximum level. */
     num_children_child = ts->t8_element_num_children (child);
-    for (j = 0; j < num_children_child; j++) {
-      ts->t8_element_child (child, j, element);
-      /*Check the computation of the successor */
-      SC_CHECK_ABORT (!ts->t8_element_compare (element, successor)
-                      , "Wrong Successor at Maxlvl\n");
-      /*Compute the next successor */
+    for (int jchild = 0; jchild < num_children_child; jchild++) {
+      ts->t8_element_child (child, jchild, element);
+      /* Check the computation of the successor. */
+      ASSERT_TRUE (!ts->t8_element_compare (element,
+                                            successor)) <<
+        "Wrong Sucessor at Maxlvl.\n";
+      /* Compute the next successor. */
       ts->t8_element_successor (successor, successor, maxlvl);
     }
     ts->t8_element_parent (child, element);
   }
 }
 
-static void
-t8_compute_successor (const int level)
-{
-  t8_element_t       *element, *successor, *child, *last;
-  t8_scheme_cxx      *scheme;
-  t8_eclass_scheme_c *ts;
-  int                 eclassi, i;
-  t8_eclass_t         eclass;
-  scheme = t8_scheme_new_default_cxx ();
-  for (eclassi = T8_ECLASS_LINE; eclassi < T8_ECLASS_COUNT; eclassi++) {
-    eclass = (t8_eclass_t) eclassi;
-    ts = scheme->eclass_schemes[eclass];
-    ts->t8_element_new (1, &element);
-    ts->t8_element_new (1, &successor);
-    ts->t8_element_new (1, &child);
-    ts->t8_element_new (1, &last);
-
-    ts->t8_element_set_linear_id (element, 0, 0);
-
-    /*Test at lower level */
-    for (i = 1; i <= level; i++) {
-      ts->t8_element_set_linear_id (successor, i, 0);
-      ts->t8_element_last_descendant (element, last, i);
-      t8_recursive_successor (element, successor, child, last, ts, i);
-    }
-    /*Test at Maxlevel */
-    ts->t8_element_set_linear_id (element, ts->t8_element_maxlevel () - 2, 0);
-    ts->t8_element_set_linear_id (successor, ts->t8_element_maxlevel (), 0);
-    t8_deep_successor (element, successor, last, ts);
-
-    ts->t8_element_destroy (1, &element);
-    ts->t8_element_destroy (1, &successor);
-    ts->t8_element_destroy (1, &child);
-    ts->t8_element_destroy (1, &last);
-  }
-  t8_scheme_cxx_unref (&scheme);
-}
-
-int
-main (int argc, char **argv)
+TEST_P (class_successor, test_recursive_and_deep_successor)
 {
-  int                 mpiret;
 #ifdef T8_ENABLE_DEBUG
   const int           maxlvl = 3;
 #else
   const int           maxlvl = 4;
 #endif
 
-  mpiret = sc_MPI_Init (&argc, &argv);
-  SC_CHECK_MPI (mpiret);
-  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
-  t8_init (SC_LP_DEFAULT);
+  t8_element_t       *element;
```

Comment:
Please move these elements into the class. They can be constructed, initialized and deconstructed during the SetUp() and the TearDown() phase of the test.